stdlib/messaging/rabbitmq/src/main/ballerina/rabbitmq/channel.bal

```diff
@@ -0,0 +1,132 @@
+
+    # Binds a queue to an exchange with the given binding key.
+    #
+    # + queueName - Name of the queue.
+    # + exchangeName - Name of the exchange.
+    # + bindingKey - Binding key used to bind the queue to the exchange.
+    # + return - Returns an error if an I/O error is encountered or nil if successful.
+    public remote extern function queueBind(string queueName, string exchangeName, string bindingKey) returns error?;
+
+    # Publishes a message. Publishing to a non-existent exchange will result in a channel-level
+    # protocol exception, which closes the channel.
+    #
+    # + routingKey - The routing key.
+    # + message - The message body.
+    # + exchange - The name of the exchange the message is published to.
+    # + return - Returns an error if an I/O error is encountered or nil if successful.
+    public remote extern function basicPublish(string routingKey = "", string exchange = "", string | byte[] message)
+                                      returns error?;
+
+    # Deletes a queue, without regard for whether it is in use or has messages on it, If the paramters ifUnused or
+    # ifEmpty is given, the queue is checked before deleting.
+    #
+    # + queueName - Name of the queue to be deleted.
+    # + ifUnused - True if the queue should be deleted only if not in use.
+    # + ifEmpty -  True if the queue should be deleted only if empty.
+    # + return - Returns error if an I/O error is encountered or nil if successful.
+    public remote extern function queueDelete(string queueName, boolean? ifUnused = false, boolean? ifEmpty = false)
+                                      returns error?;
+
+    # Deletes an exchange.
+    #
+    # + exchange - The name of the exchange.
+    # + ifUnused - True to indicate that the exchange is only to be deleted if it is unused.
+    # + return - An I/O error if an error is encountered or nil otherwise.
+    public remote extern function exchangeDelete(string exchange, boolean? ifUnused = false) returns error?;
+
+    # Purges the contents of the given queue.
+    #
+    # + queueName - The name of the queue.
+    # + return - An error if an I/O error is encountered or nil if successful.
+    public remote extern function queuePurge(string queueName) returns error?;
+};
+
+# Holds the parameters used to declare a queue.
+#
+# + queueName - The name of the queue, if not specified then autogenerated.
+# + durable - True if declaring a durable queue (the queue will survive a server restart).
+# + exclusive - True if we are declaring an exclusive queue (restricted to this connection).
+# + autoDelete - True if we are declaring an autodelete queue (server will
+# delete it when no longer in use).
+# + arguments - Other properties (construction arguments) for the queue.
+public type QueueConfiguration record {
+    string queueName;
+    boolean? durable = false;
+    boolean? exclusive = true;
+    boolean? autoDelete = true;
```

Comment:
Again, do we expect the user to set `()` for any of these fields? Applicable elsewhere too.